src/Extensions/SwiftypeSiteTreeCrawlerExtension.php

```diff
@@ -171,4 +176,21 @@ protected function getLogger(): LoggerInterface
     {
         return Injector::inst()->get(LoggerInterface::class);
     }
+
+    /**
+     * Sets the version context to Live as that's what crawlers will (normally) see
+     * the main function is to suppress teh ?stage=Live querystring
+     *
+     * @param callable $callback
+     * @return void
+     */
+    private function withVersionContext(callable $callback) {
+        Versioned::withVersionedMode(function() use ($callback) {
+            $originalReadingMode = Versioned::get_default_reading_mode();
+            Versioned::set_stage(Versioned::LIVE);
+            Versioned::set_default_reading_mode(Versioned::get_reading_mode());
+            $callback();
+            Versioned::set_default_reading_mode($originalReadingMode);
+        });
```

Comment:
Do we need to set the reading mode and everything else (honest question)? I was under the impression that `set_stage` already does this.

If so, perhaps this doesn't need to be a separate method, and you could just call it like this from within `onAfterPublish`/etc:
```php
Versioned::withVersionedMode(function (): void {
    Versioned::set_stage(Versioned::LIVE);

    $this->forceSwiftypeIndex();
});
```

Reply:
yeah the current stage needs to match the default stage to suppress the query string, and the cms will set the default to draft when viewing a page in the edit form - see https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L751